app/renderer/js/main.ts

```diff
@@ -1,53 +1,119 @@
 'use strict';
 
-const { ipcRenderer, remote, clipboard, shell } = require('electron');
-const isDev = require('electron-is-dev');
+import { ipcRenderer, remote, clipboard, shell } from 'electron';
+import isDev from 'electron-is-dev';
+import escape from 'escape-html';
+import electronConnect from 'electron-connect';
+import feedbackHolder from './js/feedback';
 
 const { session, app, Menu, dialog } = remote;
-const escape = require('escape-html');
 
-require(__dirname + '/js/tray.js');
-const DomainUtil = require(__dirname + '/js/utils/domain-util.js');
-const WebView = require(__dirname + '/js/components/webview.js');
-const ServerTab = require(__dirname + '/js/components/server-tab.js');
-const FunctionalTab = require(__dirname + '/js/components/functional-tab.js');
-const ConfigUtil = require(__dirname + '/js/utils/config-util.js');
-const DNDUtil = require(__dirname + '/js/utils/dnd-util.js');
-const ReconnectUtil = require(__dirname + '/js/utils/reconnect-util.js');
-const Logger = require(__dirname + '/js/utils/logger-util.js');
-const CommonUtil = require(__dirname + '/js/utils/common-util.js');
+// eslint-disable-next-line import/no-unassigned-import
+require('./js/tray');
 
-const { feedbackHolder } = require(__dirname + '/js/feedback.js');
+import DomainUtil = require('./js/utils/domain-util');
+import WebView = require('./js/components/webview');
+import ServerTab = require('./js/components/server-tab');
+import FunctionalTab = require('./js/components/functional-tab');
+import ConfigUtil = require('./js/utils/config-util');
+import DNDUtil = require('./js/utils/dnd-util');
+import ReconnectUtil = require('./js/utils/reconnect-util');
+import Logger = require('./js/utils/logger-util');
+import CommonUtil = require('./js/utils/common-util');
 
 const logger = new Logger({
 	file: 'errors.log',
 	timestamp: true
 });
 
+interface FunctionalTabProps {
+	name: string;
+	materialIcon: string;
+	url: string;
+}
+
+interface StringIndexType {
+	[key: string]: any;
+}
+
+interface SettingsOptions {
+	[key: string]: any;
```

Comment:
I remove this `[key: string]: any`. The reason you used it was because of the `for` loop which I solved by doing:
```typescript
for (const i in settingOptions) {
    const setting = i as keyof SettingsOptions;
    if (ConfigUtil.getConfigItem(i) === null) {
        ConfigUtil.setConfigItem(i, settingOptions[setting]);
    }
}
```